src/promptflow/tests/sdk_cli_test/unittests/test_utils.py

```diff
@@ -199,6 +202,19 @@ def test_concurrent_execution_of_refresh_connections_dir(self, concurrent_count)
         for thread in threads:
             thread.join()
 
+    @pytest.mark.parametrize("concurrent_count", [1, 2, 4, 8])
+    def test_concurrent_hint_for_update(self, concurrent_count):
+        from concurrent.futures import ThreadPoolExecutor
+        with patch('promptflow._utils.version_hint_utils.datetime') as mock_datetime:
+            mock_datetime.datetime.now.return_value = datetime.datetime.now()
+            mock_datetime.datetime.strptime.return_value = datetime.datetime.now() - datetime.timedelta(days=8)
+            mock_datetime.timedelta.return_value = datetime.timedelta(days=7)
+            hint_for_update()
+            with ThreadPoolExecutor(max_workers=concurrent_count) as pool:
```

Comment:
How does max_workers impact `check_latest_version`'s behavior?

Reply:
Actually, this is added when hint_for_update and check_latest_version need to be run in one thread pool and I need check the file is well writen in max_workers. Will update this test in later pr.